helix-core/src/main/java/org/apache/helix/task/AbstractTaskDispatcher.java

```diff
@@ -1240,16 +1240,18 @@ protected void finishJobInRuntimeJobDag(TaskDataCache clusterDataCache, String w
 
   /**
    * TODO: Move this logic to Task Framework metrics class for refactoring.
-   * Computes and passes on submissionToProcessDelay to the dynamic metric.
+   * Computes and passes on submissionToProcessDelay to the dynamic metric; syncs the metric to
+   * ZooKeeper
    * @param dataProvider
    * @param clusterStatusMonitor
    * @param workflowConfig
    * @param jobConfig
    * @param currentTimestamp
+   * @param helixManager
    */
   protected static void reportSubmissionToProcessDelay(BaseControllerDataProvider dataProvider,
       final ClusterStatusMonitor clusterStatusMonitor, final WorkflowConfig workflowConfig,
-      final JobConfig jobConfig, final long currentTimestamp) {
+      final JobConfig jobConfig, final long currentTimestamp, final HelixManager helixManager) {
```

Comment:
It's strongly recommend we check whether the given ZkConnection is valid. What if helixManager is null or not connected?